src/components/updateable/updateable.jsx

```diff
@@ -27,17 +29,13 @@ export default class Updateable extends React.Component {
     this.updateClass(this.props.value, nextProps.value);
   }
 
-  shouldComponentUpdate(nextProps, nextState) {
-    return this.state.value !== nextState.value || this.state.updateableClass !== nextState.updateableClass;
-  }
-
   updateClass(oldValue, newValue) {
     const diff = newValue - oldValue;
     if (diff === 0) return;
 
     this.setState(
       {
-        diff, // eslint-disable-line
+        diff,
```

Comment:
Maybe we can remove diff from the internal state? it seems like it is not used anyway

Reply:
I think it's optionally used in the component that this component renders with it's render prop but maybe @jesperwiner knows?

Reply:
Yes it's used outside this component. I think that this component would be much easier to understand if it was an HOC

Reply:
Oh yeah I see now in the comment below `/** `(props, {diff, updateableClass, value}=state) => <Component />` */`. Hmm ok lets go with this then :)